### PR TITLE
Use supertap to generate TAP output

### PR DESCRIPTION
**Pull request by vadimdemedes**
Uses https://github.com/vadimdemedes/supertap to generate TAP output, without changing a single line of tests and a lot of the code was extracted from AVA itself, so it's pretty much backwards compatible.

### CODE REVIEW

File: lib/reporters/tap.js [hunk 1]
```diff
@@ -41,79 +39,42 @@ class TapReporter {
...
-		}
-
-		return output.join('\n');
+		return supertap.test(test.title, {
+			passed: !test.error,
+			index: ++this.i,
+			todo: test.todo,
+			skip: test.skip,
+			comment: test.logs,
+			error: test.error ? dumpError(test.error, true) : null
```
Comment (novemberborn) [inline lib/reporters/tap.js]: `dumpError` doesn't return an `Error` instance, which `supertap`s API says is expected. Perhaps the API documentation should be loosened, or clearer on what properties are expected on non-`Error` objects?

**Comment by vadimdemedes** [comment]
> Heh, "pretty much" implies there might be some behavioral changes?

I don't think so, since I haven't modified AVA's tests and they pass 😉

File: lib/reporters/tap.js (hunk 1 above)
Comment (vadimdemedes) [reply lib/reporters/tap.js]: Agree, will clarify that.

Comment (vadimdemedes) [reply lib/reporters/tap.js]: Done.

**Pull request merged**